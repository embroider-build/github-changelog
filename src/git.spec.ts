import { parseLogMessage } from "./git";
import { describe, it, expect } from "vitest";

const messages = [
  "hash<a0000000> ref<> message<fix: some random fix which will be ignored> date<1966-01-01>",
  "hash<a0000001> ref<tag: a-new-hope@4.0.0, tag: empire-strikes-back@5.0.0, tag: return-of-the-jedi@6.0.0> message<fix: some random fix which will be ignored> date<1966-01-01>",
  "hash<a0000002> ref<tag: v0.1.0> message<Merge pull request #1 from star-wars> date<1977-01-01>",
  "hash<a0000003> ref<RC;.;0.1> message<;;fix;;;> date<20001-01-01>",
  "hash<a0000004> ref<RC;.;0.1> message<feat: add <App/>> date<20001-01-01>",
  "hash<a0000005> ref<a-new-rc<1>> message<feat(app): dev -> rc1> date<2020-01-01>",
  "hash<a0000006> ref<a-new-rc<1>> message<feat(app): dev > rc1> date<2020-01-01>",
  "hash<a0000007> ref<<>> message<<>> date<<>>",
  "hash<a0000008> ref<>> message<>> date<>>",
  // nullable results
  "hash<a0000009> ref<a-new-rc<1>> message<feat(app): dev > rc1>",
  "hash<a00000010> ref<a-new-rc<1>>",
  "hash<a0000011>",
  "hash<a0000012> date<2020-01-01>",
];

describe("parseLogMessage", function () {
  it("should return null for empty message", function () {
    expect(parseLogMessage("")).toEqual(null);
  });

  it("should return null for message with invalid format", function () {
    expect(parseLogMessage("hash<a000002 ref<%>")).toEqual(null);
  });

  messages.forEach((message, i) => {
    it("should parse message " + i, function () {
      expect(parseLogMessage(message)).toMatchSnapshot();
    });
  });
});
