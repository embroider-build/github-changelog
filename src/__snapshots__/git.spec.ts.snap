// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseLogMessage > should parse message 0 1`] = `
{
  "date": "1966-01-01",
  "refName": "",
  "sha": "a0000000",
  "summary": "fix: some random fix which will be ignored",
}
`;

exports[`parseLogMessage > should parse message 1 1`] = `
{
  "date": "1966-01-01",
  "refName": "tag: a-new-hope@4.0.0, tag: empire-strikes-back@5.0.0, tag: return-of-the-jedi@6.0.0",
  "sha": "a0000001",
  "summary": "fix: some random fix which will be ignored",
}
`;

exports[`parseLogMessage > should parse message 2 1`] = `
{
  "date": "1977-01-01",
  "refName": "tag: v0.1.0",
  "sha": "a0000002",
  "summary": "Merge pull request #1 from star-wars",
}
`;

exports[`parseLogMessage > should parse message 3 1`] = `
{
  "date": "20001-01-01",
  "refName": "RC;.;0.1",
  "sha": "a0000003",
  "summary": ";;fix;;;",
}
`;

exports[`parseLogMessage > should parse message 4 1`] = `
{
  "date": "20001-01-01",
  "refName": "RC;.;0.1",
  "sha": "a0000004",
  "summary": "feat: add <App/>",
}
`;

exports[`parseLogMessage > should parse message 5 1`] = `
{
  "date": "2020-01-01",
  "refName": "a-new-rc<1>",
  "sha": "a0000005",
  "summary": "feat(app): dev -> rc1",
}
`;

exports[`parseLogMessage > should parse message 6 1`] = `
{
  "date": "2020-01-01",
  "refName": "a-new-rc<1>",
  "sha": "a0000006",
  "summary": "feat(app): dev > rc1",
}
`;

exports[`parseLogMessage > should parse message 7 1`] = `
{
  "date": "<>",
  "refName": "<>",
  "sha": "a0000007",
  "summary": "<>",
}
`;

exports[`parseLogMessage > should parse message 8 1`] = `
{
  "date": ">",
  "refName": ">",
  "sha": "a0000008",
  "summary": ">",
}
`;

exports[`parseLogMessage > should parse message 9 1`] = `null`;

exports[`parseLogMessage > should parse message 10 1`] = `null`;

exports[`parseLogMessage > should parse message 11 1`] = `null`;

exports[`parseLogMessage > should parse message 12 1`] = `null`;
